codec-http/src/main/java/io/netty/handler/codec/http/cookie/ServerCookieDecoder.java

```diff
@@ -96,45 +96,43 @@ private ServerCookieDecoder(boolean strict) {
             }
 
             int nameBegin = i;
-            int nameEnd = i;
-            int valueBegin = -1;
-            int valueEnd = -1;
-
-            if (i != headerLen) {
-                keyValLoop: for (;;) {
-
-                    char curChar = header.charAt(i);
-                    if (curChar == ';') {
-                        // NAME; (no value till ';')
-                        nameEnd = i;
-                        valueBegin = valueEnd = -1;
-                        break keyValLoop;
-
-                    } else if (curChar == '=') {
-                        // NAME=VALUE
-                        nameEnd = i;
-                        i++;
-                        if (i == headerLen) {
-                            // NAME= (empty value, i.e. nothing after '=')
-                            valueBegin = valueEnd = 0;
-                            break keyValLoop;
-                        }
-
-                        valueBegin = i;
-                        // NAME=VALUE;
-                        int semiPos = header.indexOf(';', i);
-                        valueEnd = i = semiPos > 0 ? semiPos : headerLen;
-                        break keyValLoop;
-                    } else {
-                        i++;
-                    }
+            int nameEnd;
+            int valueBegin;
+            int valueEnd;
+
+            for (;;) {
 
+                char curChar = header.charAt(i);
+                if (curChar == ';') {
+                    // NAME; (no value till ';')
+                    nameEnd = i;
+                    valueBegin = valueEnd = -1;
+                    break;
+
+                } else if (curChar == '=') {
+                    // NAME=VALUE
+                    nameEnd = i;
+                    i++;
                     if (i == headerLen) {
-                        // NAME (no value till the end of string)
-                        nameEnd = headerLen;
-                        valueBegin = valueEnd = -1;
+                        // NAME= (empty value, i.e. nothing after '=')
+                        valueBegin = valueEnd = 0;
                         break;
                     }
+
+                    valueBegin = i;
+                    // NAME=VALUE;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)